Add on

maryamariyan opened the pull request:
I'm trying to make changes to ExternDll.Odbc32 to make it adhere to interop conventions here:
https://github.com/dotnet/corefx/blob/42ce0e696e08e190aceaf8613e47e67d57178550/Documentation/coding-guidelines/interop-guidelines.md

still work in progress. need to figure out why it's not working yet

michellemcdaniel committed:
Add PerformanceCounter Library

michellemcdaniel committed:
Port System.Diagnostics.PerformanceCounters

This change adds System.Diagnostics.PerformanceCounters to .NET Core. It
also adds two tests for the library.

michellemcdaniel committed:
Update based on feedback

michellemcdaniel committed:
Fix additional comments

michellemcdaniel committed:
Address feedback

michellemcdaniel committed:
Remove unnecessary references

michellemcdaniel committed:
Add tests, update ref, update guid

Commit:
Fix tests build

Commit:
Add baseline file

Commit:
Fix configs for VS in src

Commit:
Fix dir.props to 4.0.0.0

Commit:
Fix key

michellemcdaniel committed:
Fix feedback

michellemcdaniel committed:
Fix based on feedback

michellemcdaniel committed:
Remove duplicate string from resource file

michellemcdaniel committed:
Fix csproj

michellemcdaniel committed:
Fix tests for UAP

michellemcdaniel committed:
Fix IsAppContainerProcess for Windows 7

michellemcdaniel committed:
Fix IsAppContainerProcess

michellemcdaniel committed:
Fix formatting

michellemcdaniel committed:
update the CreateCounter test to use custom counter

michellemcdaniel committed:
Fix csproj for linux/osx

michellemcdaniel committed:
fix netfx ref build

michellemcdaniel committed:
Address feedback, netfx build

michellemcdaniel committed:
Fix netfx, comments

michellemcdaniel committed:
Update csproj, sln, and dirs.props for PerfCounter

michellemcdaniel committed:
Address Feedback

michellemcdaniel committed:
Fix issues

michellemcdaniel committed:
Move ProcessWaitHandle to Interop

michellemcdaniel committed:
add ProcessWaitHandle.cs

safern commented on src/Common/src/Interop/Linux/Interop.Libraries.cs:
```diff
@@ -0,0 +1,12 @@
+{
+    private static partial class Libraries
+    {
+        internal const string Odbc32 = "libodbc.2.dylib";
```
You could instead add this in Interop/Unix/Interop.Libraries.cs I see that both OSX and this one are the exact same value.

safern commented on src/Common/src/Interop/OSX/Interop.Libraries.cs:
```diff
@@ -11,6 +11,7 @@ private static partial class Libraries
         internal const string libproc = "libproc";
         internal const string LibSystemCommonCrypto = "/usr/lib/system/libcommonCrypto";
         internal const string LibSystemKernel = "/usr/lib/system/libsystem_kernel";
+        internal const string Odbc32 = "libodbc.2.dylib";
```
Once you add it into the Unix file you can delete this one.

maryamariyan replied:
My mistake. It's supposed to be different for linux and that's why I added the new file

safern commented on src/System.Data.Odbc/src/System.Data.Odbc.csproj:
```diff
@@ -111,13 +111,19 @@
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetsOsx)' == 'true' And '$(IsPartialFacadeAssembly)' != 'true' ">
-    <Compile Include="Common\System\Data\Common\ExternDll.Osx.cs" />
+    <Compile Include="$(CommonPath)\Interop\OSX\Interop.Libraries.cs">
```
Once you unify the change into the Unix file, you can delete both ItemGroups the TargetsOsx and TargetsLinux ones and then only use one with TargetsUnix and include the file there, i.e: 

```xml
<ItemGroup Condition=" '$(TargetsUnix)' == 'true' ">
   <Compile Include="$(CommonPath)\Interop\Unix\Interop.Libraries.cs">
       <Link>Common\Interop\Unix\Interop.Libraries.cs</Link>
    </Compile>
</ItemGroup>
```

Also, `IsPartialFacadeAssembly` is only set to true when `TargetGroup` == `netfx` which can't be built on non windows environments, so it wouldn't be needed for this `ItemGroup` as a condition.

safern commented on src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs:
```diff
@@ -72,6 +72,32 @@ public void ProcessStart_UseShellExecute_OnLinux_ThrowsIfNoProgramInstalled()
+
+            Console.WriteLine($"({nameof(ProcessStart_UseShellExecute_DirectoryAndFileWithSameName_Success)}) fileToOpen: {fileToOpen} and folderToOpen: {folderToOpen}" );
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || programToOpen != null)
```
`PlatformDetection.IsOSX` instead of RuntimeInformation...

safern reviewed (commented):
Left some comments

dotnet-maestro-bot committed:
Update Standard to preview1-25811-01 (#24573)

michellemcdaniel committed:
Remove Environment Checks from SharedUtils

michellemcdaniel committed:
Merge pull request #24335 from adiaaida/addPerformanceCounters

Port System.Diagnostics.PerformanceCounters to .NET Core

michellemcdaniel committed:
Revert "Port System.Diagnostics.PerformanceCounters to .NET Core"

michellemcdaniel committed:
Merge pull request #24578 from dotnet/revert-24335-addPerformanceCounters

Revert "Port System.Diagnostics.PerformanceCounters to .NET Core"

safern committed:
Disable System.Drawing.Common crashing test (#24581)

michellemcdaniel committed:
Merge pull request #24579 from adiaaida/addPerformanceCounters

Port System.Diagnostics.PerformanceCounters to .NET Core

janvorli committed:
Update CentOS 6 docker image (#24576)

This update fixes problem with git not supporting https protocol

justinvp committed:
Avoid StringBuilder/int[] overhead with GetComputerName (#24589)

Avoid the `StringBuilder` allocation and marshalling overhead as well as
the unnecessary `int[]` allocation when p/invoking `GetComputerName`
from `PerformanceCounterLib`.

stephentoub committed:
Avoid generic interface GetEnumerator empty allocations in immutable collections (#24565)

* Avoid generic interface GetEnumerator empty allocations in immutable collections

* Address PR feedback

dotnet-maestro-bot committed:
Update CoreFx, CoreSetup to preview1-25812-01, preview1-25811-02, respectively (#24595)

dotnet-maestro-bot committed:
Update ProjectNTfs, ProjectNTfsTestILC to beta-25812-00, beta-25812-00, respectively (#24596)

KostaVlev committed:
Adding System.Collections.NonGenerics ArrayList performance tests. (#18765) (#24500)

* Adding System.Runtime.Extensions ArrayList performance tests.

* PR feedback - Moving Perf.ArrayList.cs to System.Collections.NonGenerics\tests\Performance and some optimizations.

* Spelling corrections.

* PR feedback.

* Small optimizations - removing duplicated variables.

* Fixing RemoveRange test setup. Setting more Inner iterations for the Add test.

dnickless committed:
Linq.SequenceEqual: Improved performance for collections that support index based access (#23368)

* Improved performance for collections that support index based access.

* Cast to invariant interface IList<T> instead for performance reasons + cache iteration count

dotnet-maestro-bot committed:
Update CoreFx, Standard to preview1-25812-02, preview1-25812-01, respectively (#24605)

maryamariyan committed:
improve messaging when odbc dependency is not installed (#24535)

Fixes a part of improvements listed in #17405

DrewScoggins committed:
Fix call to create submission metadata (#24609)

We made a change to the Benchview tooling and did not update the use of
it in the CoreFX repo.  This fixes that.

tarekgh committed:
Directory Services Tests - 4 (#24590)

The added test is mainly to test the interaction with the native COM interfaces.

maryamariyan committed:
work in progress. making it compile
